### PR TITLE
Backpack Re-balance MK II 

### PR DESCRIPTION
**Pull request by DaveTheHeadcrab**
🆑 Dave
tweak: Backpacks and duffelbags may no longer be accessed while they are equipped.
tweak: Satchels have reduced storage capacity (Half of a backpack rounded down).
/🆑

Re-submission of my previous PR after I royally fucked up using rebase late at night.
All storage items have a new: block_open_while_equipped. If the items have slot_flags and this var is set, the storage item will not allow items to be removed if it is found within the contents of the opening mob's get_all_slots() list. **Note:** You can still see the contents of these bags, just not touch them.

Backpacks and duffel bags have block_open_while_equipped set.

Satchels do not have block_open_while_equipped, however their storage capacity has been altered from the standard max_combined_w_class and storage_slots value of 21 to a new value of 9. This means that a satchel could fit 3 'normal' sized items in it, and up to 9 'tiny' items in it.

Rewrites a bit of code for how storage slots are displayed. Storage items will now display slots up to the number of tiny items it can fit into itself so long as it doesn't go past the first row. This is so players can get a better understanding of how much space their storage actually has. Images of this in action:

satchel1
satchel2
satchel3
satchel4
satchel5

This only works on single-row storage items currently due to how the storage display works.

Refactors `get_all_slots` to be defined at the carbon level.

Adds `BlockReach` to click code, a proc defined at the atom level that will block an attempt to click on an item if TRUE is returned.

Adds the `sum_w_class()` proc to storage objects, as this functionality was used in multiple places in the storage.dm code. Consolidated into a single proc for easy use.

Reduces the amount of storage capacity for survival boxes by 3 slots. (To 4 slots in total now)

For the sake of preserving the conversation on the previous PR: https://github.com/OracleStation/OracleStation/pull/351

### CODE REVIEW

**Comment by Spartan6** [comment]
Perhaps 14 would be better for satchels. Satchels are not really small enough that they can only fit 9 pens inside of it, for example.

**Comment by DaveTheHeadcrab** [comment]
1) If storage sizes are to be adjusted, increments of 3 are best (So that it fits an even number of "normal" sized items)
2) I'd prefer to see how this runs during tests before adjusting numbers.

**Comment by EldritchSigma** [comment]
So uh question, currently "theorycrafting" on what you can carry on certain roles that you need to carry right now: Do 4 tiny items equal one large item, 3 tiny items a normal sized item, 2 tiny items a small item?

**Comment by DaveTheHeadcrab** [comment]
Tested locally, got all the functionality I wanted out of this (And more than I planned to, honestly).
This is in a merge-ready state.

File: code/game/objects/items/storage/backpack.dm [hunk 1]
```diff
@@ -101,6 +102,7 @@
 	w_class = WEIGHT_CLASS_BULKY
 	max_w_class = WEIGHT_CLASS_NORMAL
 	max_combined_w_class = 60
+	block_open_while_equipped = FALSE // Santa needs to be fast.
```
Comment (FlattestGuitar) [inline code/game/objects/items/storage/backpack.dm]: Santa power creep. We've talked about this, Dave.

**Comment by DaveTheHeadcrab** [comment]
PR updated as per Flattest's request: Smaller survival boxes are now visually distinct (But still compatible with the regular box's overlays).
Also refactored survival boxes, because evidently I have not refactored enough shit in this PR quite yet.

**Comment by DaveTheHeadcrab** [comment]
Closed and re-opened, trying to kick travis into working properly.

**Comment by Joseph-Case** [comment]
:thumbsdown: x :100:

**Comment by Birdtalon** [comment]
Horrible change. Being able to inventory manage fast is an essential gameplay feature and removing that is just adding pointless clicking and removing the ability to quickly access your items. 👎

**Comment by SavinienBrassheart** [comment]
This is a very ill conceived and poorly thought of pull request. Imagine trying to save a life, to get that medkit in your back pack you will have to take it off, get the med kit out, put the backpack on or drop it,  oh that hard crit person is dead.

Or imagine being antag and sec is chasing you, you got your double e-sword but it’s in your backpack. “Hey sec can you stop chasing me so I can get my weapon and kill you? Just gotta take my backpack off”

This pull request doesn’t improve enjoyment of the game, it does not add a fun little challenge, nor does it balance anything. It will simply ruin oracle and any hope of having fun on it. I ask you to go around to the different big servers and ask the players what they think of this idea. Yogg, hippie, paradise, and colonial marines, to name a few. Ask their players if they would come dashing to your server with this ridiculous PR in place.

Some will laugh, believing the PR is a joke and that no one could really think it’s a good idea.

Some will shake their heads in disbelief over such a terrible concept.

But none will approve of it, none will ask their admins to bring this to their server.

So please, if you want oracle to have any chance of being a successful server. Delete this pull request and leave the storage system alone.

**Comment by AndrewMontagne** [comment]
Someone disliked this PR enough to register on GitHub to comment on it. CONTROVERSIAL.

**Comment by Ziiro** [comment]
I've softened a bit on the idea since my last post. However my stance remains: I'd be willing to try it, but I have my doubts about it in general.

I understand what you're doing here, but these are the kinds of things you allow a little leeway on for the sake of the game being enjoyable rather than a hassle. But who knows? It's an interesting idea. Could be okay.

**Comment by Joseph-Case** [comment]
Can this get the controversial PR label

**Comment by AndrewMontagne** [comment]
@Joseph-Case Uh, it already has?

**Comment by Joseph-Case** [comment]
I was using a mobile client which didnt show tags, sorry

**Comment by Cid777** [comment]
@Birdtalon Use a satchel?

People are _used_ to having a large inventory, but you really don't need it. This alleviates the problem of walking armories, and makes it more difficult to surprise someone with a weapon they couldn't see coming. It makes the game less arcade-ey.

At best, people consider their inventory and bag choice more carefully, and at worst they're annoyed for a while?

**Comment by Birdtalon** [comment]
I consider the game being arcade-ey a good thing. Inventory management and building up a plethora of items to use is good. The argument that it forces people to think about their kit and tactics more is all well and good but trying to force realism and tactics on a game which really doesn't support it just turns into an unsatisfying mess.

**Comment by EldritchSigma** [comment]
Alright thought about this a little bit... the main issue are walking armories right? Then why not target that instead, I checked most weapons... they are normal sized. If you up the size by 1, that means making them large sized instead of normal sized you cannot fit that many guns into your backpack, at most 2 or 3, 3 only with significantly reduced storage capacity. The Bluespace bagpack would fit still alot of shitty guns though, but that thing has one significant drawback. If you teleporter you most likely die. Any form of teleportation triggers it. Then we have duffelbacks which could still carry bulk guns but those slow you down already as a thick big disadvantage. About people "hording" stuff like tools. Its a spacestation. If I were working on a space station you can bet your damn ass on that I would carry a full toolset at any given time. Engineers do major repairing, sure. But anyone should atleast have a basic set of tools to open depowered doors or fix a minor power issue. Its a space station, its a high end, highly secure facility, you could expect people to be able to do maintenance to a certain degree regardless of their profession.

 I mean its silly that people carry a stack of metal arround together with a stack of glass. But rather adjust that so they are not normal sized but ATLEAST large, if not huge and give engineers a special piece of equipment to carry metal and glass arround for repairs. But its totally manageable to put an entire toolset, including a small emergency oxygen tank and a box with emergency medication into your satchel without any issues. Its also entirely possible to access all items from your backpack without taking it off IRL. Aslong you dont have 2 same sized and shaped items or similar weight objects you can get them out pretty well. What would be less intrusivel is a slowdown whenever you open your back/satchel since you need to focus on getting the items out since noone is fully capable of multitasking (except you have some very rare form of mental constitution which allows the left half of the brain to operate individually from the right side). 

All of this combined would be less of a pain in the ass to manage for people, does nerf backpacks so they are not as "gamy" and would avoid walking armories from happening every shift. Heck I would be even down for a weight system. The more you carry the more it could slow you down... instead of this thing that implies that your satchel is the size of a tiny bowl which ruins immersion harder than it helps creating (atleast for me).

**Comment by Cid777** [comment]
> I consider the game being arcade-ey a good thing.

Inherently?

> Inventory management and building up a plethora of items to use is good.

... **_Inherently?_**

**Comment by Joseph-Case** [comment]
Add stale label

**Comment by SamHPurp** [comment]
@Joseph-Case lets chill out.

**Comment by SmokingKilz** [comment]
Though I personally don't think these changes would work for Oracle, I'm not sure if the level of some of these reactions are warranted, people bashing the system saying it'd never work needs to take a moment to go check out Colonial Marines, because it definitely can work. Some of the points brought up are frankly a little over dramatic to put it lightly.

So saying that these changes are "poorly thought of" is definitely just factually not correct. Now to actually look at this PR properly, I think it would be interesting but I don't think it'd fit Oracle like it fits CM, just my personal opinion.

**Comment by SamHPurp** [comment]
This was an interesting test merge, sadly it went how I thought it would.

This isn't going to work for us I think, I loved the idea, and but even I don't enjoy the actuality of it.

**Pull request closed without merging**